calendar/src/main/java/com/minuk/calendar/PetCalendarView.kt

```diff
@@ -0,0 +1,304 @@
+    private var headerViewHeight = 0f
+    private var itemViewRatio = DEFAULT_ITEM_DIMENS_RATIO
+
+    //Ui data property
+    private var weekendColor = DEFAULT_WEEKEND_COLOR
+    private var weekdaysColor = DEFAULT_WEEKDAYS_COLOR
+    private var todayColor = DEFAULT_TODAY_COLOR
+
+    private var eventHandler: PetCalendarEventHandler? = null
+
+    private var calendarAdapter: PetCalendarAdapter? = null
+
+    init {
+        initLayout(context)
+        initHeaderSize()
+
+        initCalendarData()
+        initDayRecyclerView()
+    }
+
+    private fun initLayout(context: Context) {
+        val inflater = LayoutInflater.from(context)
+
+        calendarBinding = ViewCalendarBinding.inflate(inflater)
+
+        this.addView(calendarBinding.root)
+    }
+
+    private fun initHeaderSize() {
+        headerViewHeight = resources.getDimension(R.dimen.height_header)
+    }
+
+    private fun initCalendarData() {
+        //Log.d(TAG, "year: ${calendar[Calendar.YEAR]} month: ${calendar[Calendar.MONTH]}")
+
+        toDay = calendar[Calendar.DAY_OF_MONTH]
+        calendar.set(Calendar.DAY_OF_MONTH, 1)
+
+        startDayOfWeek = calendar[Calendar.DAY_OF_WEEK] - 1
+        endDayOfCurrentMonth = calendar.getActualMaximum(Calendar.DAY_OF_MONTH)
+
+        calendar.add(Calendar.DAY_OF_MONTH, -1)
+        endDayOfLastMonth = calendar.getActualMaximum(Calendar.DAY_OF_MONTH)
+
+        daysCount = startDayOfWeek + endDayOfCurrentMonth
+        if (daysCount % 7 != 0) {
+            daysCount += 7 - daysCount % 7
+        }
+
+        setItemViewDimensRadio()
+
+        if (calendarAdapter == null) {
+            calendarAdapter = PetCalendarAdapter()
+        } else {
+            calendarAdapter?.notifyDataSetChanged()
+        }
+    }
+
+    private fun initDayRecyclerView() {
+        calendarBinding.daysRecyclerview.apply {
+            setItemViewCacheSize(10)
+            adapter = calendarAdapter
+        }
+    }
+
+    /**
+     * 측정된 가로와 세로 길이로 ratio를 계산하여
+     * 이전 ratio와 다르면 item view 재생성
+     */
+    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+
+        when (MeasureSpec.getMode(widthMeasureSpec)) {
+            MeasureSpec.UNSPECIFIED -> viewWidth = widthMeasureSpec
+            MeasureSpec.EXACTLY -> viewWidth = MeasureSpec.getSize(widthMeasureSpec)
+        }
+
+        when (MeasureSpec.getMode(heightMeasureSpec)) {
+            MeasureSpec.UNSPECIFIED -> viewHeight = heightMeasureSpec
+            MeasureSpec.EXACTLY -> viewHeight = MeasureSpec.getSize(heightMeasureSpec)
+        }
+
+        setItemViewDimensRadio(true)
+    }
+
+    private fun setItemViewDimensRadio(update: Boolean = false) {
+        val ratio = getItemViewDimensRatio()
+
+        if (itemViewRatio != ratio) {
+            itemViewRatio = ratio
+
+            if (update) {
+                calendarAdapter?.notifyDataSetChanged()
+            }
+        }
+    }
```

Comment:
```suggestion
            itemViewRatio = ratio

            if (update) {
                calendarAdapter?.notifyDataSetChanged()
            }
        }
```
depth 를 줄이면서, 동일한 코드인 것 같아 제안드려봅니다😊

Reply:
감사합니다!